frontend/src/components/recipepage.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect } from 'react';
+
+import './style.css'
+
+export default function RecipePage({name, imagesrc, ingredients, procedure}) {
+
+    const [newIngredient, setNewIngredient] = useState('');
+    const [newInstruction, setNewInstruction] = useState('');
+    const [addIngredient, setAddIngredient] = useState('');
+    const [addInstruction, setAddInstrction] = useState('');
+
+
+    useEffect(() => {
```

Comment:
You don't need to have a useEffect for POST requests. Remember, By using the useEffect Hook, you tell React that your component needs to do something after render.

However, the POST request should be ran whenever the addIngredient or addInstruction button is clicked, so we could just have the fetch be inside of a function that runs when the button is clicked (But it looks like you got it working this way so whatever works works!!)